### DISCUSSION TITLE
Search Read Messages

### DISCUSSION THREAD

**Discussion by timothymamo** [ahochsteger/gmail-processor#389] [Q&A] [answered]
I'm getting started with `gmail-processor` and testing out a simple script I'd like to implement.

I'm trying to do a simple search for __read__ messages with an attachment between specific dates. I then want to label these processed messages, with `gmail-processed` so that the script doesn't process them again. Below is the config I'm using:

```
function run() {
  const config = {
    description:
      "Use this to get started with Gmail Processor. It stores an attachment to Google Drive.",
    settings: {
      markProcessedMethod: "mark-read",
    },
    global: {
      thread: {
        match: {
          query:
            'has:attachment -in:trash -in:drafts -in:spam -label:gmail-processed is:read after:${date.now:date:startOfQuarter-1d:yyyy-MM-dd} before:${date.now:date:endOfQuarter+1d:yyyy-MM-dd} subject:"^*word*$"',
        },
      },
    },
    threads: [
      {
        match: {
          query: 'to:sample@email.com is:read',
        },
        messages: [
          {
            match: {
              query: 'is:read',
            },
            attachments: [
              {
                match: {
                  name: "^.*\.pdf$",
                },
                actions: [
                  {
                    name: "attachment.store",
                    args: {
                      location:
                        "/GmailProcessor-Tests/e2e/simple/${message.date:date::yyyy-MM-dd}/${message.subject}-${attachment.name}",
                      conflictStrategy: "keep",
                    },
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  }

  return GmailProcessorLib.run(config, "dry-run")
}
```

After running the script, the process _does_ find the email I'm looking for but outputs the following `DEBUG` message:

```
1:28:46 PM Info 2024-07-31T11:28:46.931Z TRACE [ThreadProcessor.processEntity()] Processing thread id '123...456' (lastMessageDate:2024-07-31T08:26:36.000Z, subject:'word') started ...
1:28:46 PM Info 2024-07-31T11:28:46.932Z TRACE [MessageProcessor.processConfigs()] Processing message configs started ...
1:28:46 PM Info 2024-07-31T11:28:46.934Z TRACE [MessageProcessor.processConfig()] Processing message config '0' started ...
1:28:47 PM Info 2024-07-31T11:28:47.047Z DEBUG NO MATCH: message is read
1:28:47 PM Info 2024-07-31T11:28:47.053Z INFO Skipping non-matching message id '123...456' (date:'2024-07-31T08:26:36.000Z',  subject:'word', from:"email>).
```

Meaning the subsequent actions are not performed.

I've tested the above, without the `is:read` option, by setting the email to being unread and works perfectly, both in `dry-run` and in `safe-mode`.

What am I missing? Is this possible?

P.S. For further context I want to run this script every quarter, only, for accounting purposes, meaning I'll be reading the message when I receive it (mainly to check if Its an invoice I need to pay or has been paid).

**Comment by ahochsteger** [comment, marked as answer]
@timothymamo the reason is that the config uses `mark-read` as the `markProcessedMethod` which implies that certain matches are automatically added to the configuration, since this is normally what a typical user might expect.

You may overcome this limitation if you use `custom` instead, but then you have to add the action to mark processed messages yourself.

See here for the documentation of the available options:
https://ahochsteger.github.io/gmail-processor/docs/reference/enum-types#markprocessedmethod

**Reply by timothymamo** [reply]
Ahhhh thanks for the quick reply @ahochsteger
